components/stimulation/StimulationStudioDragAndDropPanel.vue

```diff
@@ -76,6 +123,12 @@ export default {
       setting_type: "Current",
       reopen_modal: null,
       shift_click_img_idx: null,
+      shift_click_nested_img_idx: null,
+      repeat_modal: null,
+      repeat_idx: null,
+      current_number_of_repeats: null,
+      cloned: false,
+      new_cloned_idx: null,
```

Comment:
Just a lot of state management for the repeat and handling of when each modal should appear, will try and find a better way to do this